src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -192,12 +194,13 @@ internal override bool IsExpressionBodied
             MethodKind methodKind,
             bool isAutoPropertyAccessor,
             DiagnosticBag diagnostics) :
-            base(containingType, syntax.GetReference(), syntax.Body?.GetReference(), location)
+            base(containingType, syntax.GetReference(), syntax.Body?.GetReference() ?? syntax.ExpressionBody?.GetReference(), location)
         {
             _property = property;
             _explicitInterfaceImplementations = explicitInterfaceImplementations;
             _name = name;
             _isAutoPropertyAccessor = isAutoPropertyAccessor;
+            _isExpressionBodied = (syntax.Body == null && syntax.ExpressionBody != null) ? true : _property.IsExpressionBodied;
```

Comment:
If I understand you correctly, I believe you are right and, by construction, this symbol cannot be associated with a property that has IsExpressionBodied == true. That is why the old implementation was fine. The new code that you are adding, however, is misleading. So, here is what I think we should do about this line:
- Debug.Assert that _property.IsExpressionBodied is false.
- Change the line to ```_isExpressionBodied = (syntax.Body == null && syntax.ExpressionBody != null);```